File: frontend/src/api/ai.ts
import api from './client'
import type { NLTaskParseResult, SubTaskItem, AISettingsData } from '../types'

export async function parseNaturalLanguage(text: string): Promise<NLTaskParseResult> {
  const res = await api.post('/ai/parse-task', { text })
  return res.data
}

export async function suggestTags(title: string, description?: string): Promise<string[]> {
  const res = await api.post('/ai/suggest-tags', { title, description })
  return res.data.tags
}

export async function recommendPriority(title: string, description?: string): Promise<{ priority: string; reason: string }> {
  const res = await api.post('/ai/recommend-priority', { title, description })
  return res.data
}

export async function breakdownTask(title: string, description?: string): Promise<SubTaskItem[]> {
  const res = await api.post('/ai/breakdown-task', { title, description })
  return res.data.subtasks
}

export async function summarizeTasks(taskIds?: number[]): Promise<string> {
  const res = await api.post('/ai/summarize-tasks', { task_ids: taskIds })
  return res.data.summary
}

export async function detectSimilar(taskId: number): Promise<{ task_id: number; title: string; similarity_reason: string }[]> {
  const res = await api.post('/ai/detect-similar', { task_id: taskId })
  return res.data.similar_tasks
}

export async function getAISettings(): Promise<AISettingsData> {
  const res = await api.get('/ai/settings')
  return res.data
}

export async function updateAISettings(provider: string, apiKey: string): Promise<AISettingsData> {
  const res = await api.put('/ai/settings', { provider, api_key: apiKey })
  return res.data
}

File: backend/app/api/ai.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from app.database import get_db
from app.schemas.ai import (
    NLTaskParseRequest,
    NLTaskParseResponse,
    TagSuggestionRequest,
    TagSuggestionResponse,
    PriorityRequest,
    PriorityResponse,
    BreakdownRequest,
    BreakdownResponse,
    SummarizeRequest,
    SummarizeResponse,
    SimilarRequest,
    SimilarResponse,
    AISettingsResponse,
    AISettingsUpdate,
)
from app.services import ai_service

router = APIRouter()


@router.get("/settings", response_model=AISettingsResponse)
def get_settings(db: Session = Depends(get_db)):
    return ai_service.get_ai_settings(db)


@router.put("/settings", response_model=AISettingsResponse)
def update_settings(data: AISettingsUpdate, db: Session = Depends(get_db)):
    return ai_service.update_ai_settings(db, data)


@router.post("/parse-task", response_model=NLTaskParseResponse)
async def parse_task(data: NLTaskParseRequest, db: Session = Depends(get_db)):
    return await ai_service.parse_natural_language(db, data.text)


@router.post("/suggest-tags", response_model=TagSuggestionResponse)
async def suggest_tags(data: TagSuggestionRequest, db: Session = Depends(get_db)):
    return await ai_service.suggest_tags(db, data.title, data.description)


@router.post("/recommend-priority", response_model=PriorityResponse)
async def recommend_priority(data: PriorityRequest, db: Session = Depends(get_db)):
    return await ai_service.recommend_priority(db, data.title, data.description)


@router.post("/breakdown-task", response_model=BreakdownResponse)
async def breakdown_task(data: BreakdownRequest, db: Session = Depends(get_db)):
    return await ai_service.breakdown_task(db, data.title, data.description)


@router.post("/summarize-tasks", response_model=SummarizeResponse)
async def summarize_tasks(data: SummarizeRequest, db: Session = Depends(get_db)):
    return await ai_service.summarize_tasks(db, data.task_ids)


@router.post("/detect-similar", response_model=SimilarResponse)
async def detect_similar(data: SimilarRequest, db: Session = Depends(get_db)):
    return await ai_service.detect_similar(db, data.task_id)
